android/src/main/java/com/stripeterminalreactnative/listener/RNDiscoveryListener.kt

```diff
@@ -1,20 +1,22 @@
 package com.stripeterminalreactnative.listener
 
+import com.facebook.react.bridge.Promise
 import com.facebook.react.bridge.ReactApplicationContext
-import com.stripe.stripeterminal.external.callable.Callback
 import com.stripe.stripeterminal.external.callable.DiscoveryListener
 import com.stripe.stripeterminal.external.models.Reader
 import com.stripe.stripeterminal.external.models.TerminalException
 import com.stripeterminalreactnative.ReactExtensions.sendEvent
 import com.stripeterminalreactnative.ReactNativeConstants
+import com.stripeterminalreactnative.callback.NoOpCallback
 import com.stripeterminalreactnative.createError
 import com.stripeterminalreactnative.mapFromReaders
 import com.stripeterminalreactnative.nativeMapOf
 
-class RNDiscoveryListener(
+internal class RNDiscoveryListener(
     private val context: ReactApplicationContext,
+    promise: Promise,
     private val onDiscoveredReaders: (readers: List<Reader>) -> Unit,
-) : DiscoveryListener, Callback {
+) : DiscoveryListener, NoOpCallback(promise) {
```

Comment:
This might be better as a delegate: `Callback by NoOpCallback(promise)` rather than opening and extending the class.

Reply:
Sounds good. I created a delegate variable on the class. Thanks for the suggestion!